packages/block-library/src/media-text/edit.js

```diff
@@ -187,6 +191,35 @@ function MediaTextEdit( { attributes, isSelected, setAttributes } ) {
 	const onVerticalAlignmentChange = ( alignment ) => {
 		setAttributes( { verticalAlignment: alignment } );
 	};
+
+	const { imageSizes } = useSelect( ( select ) => {
+		const { getSettings } = select( 'core/block-editor' );
+		return pick( getSettings(), [ 'imageSizes' ] );
+	} );
+	const imageSizeOptions = map(
+		filter( imageSizes, ( { slug } ) =>
+			get( image, [ 'media_details', 'sizes', slug, 'source_url' ] )
+		),
+		( { name, slug } ) => ( { value: slug, label: name } )
+	);
+	const updateImage = ( newMediaSizeSlug ) => {
+		const newUrl = get( image, [
+			'media_details',
+			'sizes',
+			newMediaSizeSlug,
+			'source_url',
+		] );
```

Comment:
Same as https://github.com/WordPress/gutenberg/pull/24795/files#r478674886

```suggestion
		const newUrl = image?.media_details?.sizes?.[ newMediaSizeSlug ]?.source_url;
```